File: core/feature06.go
package main

import (
	"math"
)

func feature06(arguments map[string]interface{}) map[string]interface{} {
	parameters := extractParameters(arguments, []string{"D", "A", "T"})
	if parameters == nil {
		return nil
	}
	D := getParameter(parameters, "D")
	A := getParameter(parameters, "A")
	T := getParameter(parameters, "T")
	result := make(map[string]interface{})
	Rad := math.Pi / 180.0
	U := float64(30)
	if D > 300 {
		U = 40
	}
	r := D / 2.0
	B := A / 2.0
	L := r * math.Tan(B*Rad)
	result["S"] = 2.0 * math.Pi * r / T
	result["L"] = []interface{}{
		2.0 * math.Pi * r,
		2.0*math.Pi*r + U,
	}
	p := make([]interface{}, 0)
	for X := float64(0); X <= T/2.0; X += 1.0 {
		F := X * (360.0 / T)
		Q := L - r*math.Tan(B*Rad)*math.Cos(F*Rad)
		p = append(p, Q)
	}
	result["Points"] = p
	return result
}

File: core/feature06.js
function feature06() {
  var parameters = extractParameters(["D", "A", "T"]);
  return feature06_core(parameters);
}

function feature06_core(parameters) {
  if (parameters == null) {
    return null;
  }
  var D = getParameter(parameters, "D");
  var A = getParameter(parameters, "A");
  var T = getParameter(parameters, "T");
  var result = {};
  var Rad = Math.PI / 180.0;
  var U = 30;
  if (D > 300) {
    U = 40;
  }
  var r = D / 2.0;
  var B = A / 2.0;
  var L = r * Math.tan(B * Rad);
  result["S"] = 2.0 * Math.PI * r / T;
  result["L"] = [
    2.0 * Math.PI * r,
    2.0 * Math.PI * r + U,
  ];
  var p = [];
  for (var X = 0; X <= T / 2.0; X += 1.0) {
    var F = X * (360.0 / T);
    var Q = L - r * Math.tan(B * Rad) * Math.cos(F * Rad);
    p.push(Q);
  }
  result["Points"] = p;
  return result;
}
